src/lsp.ts

```diff
@@ -57,7 +57,7 @@ function createClient(jarPath: string): LanguageClient {
             async provideDefinition(document, position, token, next) {
                 const nReplDefinition = await provideClojureDefinition(document, position, token);
                 if (nReplDefinition) {
-                    return nReplDefinition;
+                    return null;
```

Comment:
Shouldn't we instead check for a nrepl connection here? It seems with this approach, provideClojureDefinition will be called twice. It's not a big deal, but it is duplicated work.

Reply:
Then we would loose local definition navigation when nrepl connects. Nrepl doesn't provide that.

Reply:
I'm a bit confused, because with this change `provideClojureDefinition` is called twice, once here and once from the ClojureDefinitionProvider that we still have registered. I'm not sure why we have it registered, the comment above it says it's not used.

If we remove the ClojureDefinitionProvider (or don't register it), then return `nReplDefinition` here, then `provideClojureDefinition` will only be called once, and we'll get a definition whether nrepl is connected or not.

Reply:
Nevermind, I see now there was an issue with not using that other provider, but there is still a comment above it saying it's not used, when it must be used now, right? :thinking:

Reply:
Yes, the comment is confusing. And the whole setup is confusing, so we should try to factor it in a way that we understand what is going on. 😎

Reply:
https://github.com/BetterThanTomorrow/calva/pull/919 I tried the change I mentioned and it doesn't seem that we lose anything. Please check it out, and let me know if I'm missing something. Maybe I'm misunderstanding what you mean by "local definition navigation."